packages/engine-server/src/DendronEngineV3.ts

```diff
@@ -1318,7 +1318,7 @@ export class DendronEngineV3 extends EngineV3Base implements DEngine {
           alias = undefined;
         }
         // for user tag links, we'll have to regenerate the alias
-        if (newLoc.fname.startsWith(USERS_HIERARCHY)) {
+        if (link.type !== "ref" && newLoc.fname.startsWith(USERS_HIERARCHY)) {
```

Comment:
Can you add a comment that syntax like !@john doesn't work as a note ref, which is why we need to due `link.type !== ref`

Reply:
Also, instead of `!@john`, presently it was replaced with `@john`